tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -1,170 +1,154 @@
+  const render = ({
+    children = defaultChildText,
+    message = 'some warning message',
+    id = 'Some-confirm-button',
+    onConfirm = jest.fn(),
+    ...otherProps
+  } = {}) => {
+    return defaultRender(
+      <ConfirmButton
+        onConfirm={onConfirm}
+        message={message}
+        id={id}
+        {...otherProps}
+      >
+        {children}
       </ConfirmButton>,
-      ConfirmButtonBase,
     );
   };
 
-  const renderWithDialog = ({
-    store = dispatchClientMetadata().store,
-    ...otherProps
-  } = {}) => {
-    const root = render({ store, ...otherProps });
+  const renderWithDialog = ({ ...otherProps } = {}) => {
+    render(otherProps);
 
     // Click to open ConfirmationDialog.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find(ConfirmationDialog)).toHaveLength(1);
-
-    return root;
-  };
-
-  const getDialogProp = (root, propName) => {
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp(propName);
-    return confirmDialog.prop(propName);
+    userEvent.click(
+      screen.getByRole('button', {
+        name: defaultChildText,
+      }),
+    );
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
   };
 
   it('renders a button', () => {
-    const root = render();
+    render();
 
-    expect(root).toHaveClassName('ConfirmButton');
-    expect(root.find(Button)).toHaveLength(1);
-    expect(root.find(Button)).toHaveClassName('ConfirmButton-default-button');
-    expect(root.find(Button)).toHaveProp('buttonType', 'neutral');
+    const button = screen.getByRole('button', {
+      name: defaultChildText,
+    });
+    expect(button).toHaveClass('ConfirmButton-default-button');
+    expect(button).toHaveClass('Button--neutral');
   });
 
   it('passes the buttonType prop to the button', () => {
-    const buttonType = 'alert';
-    const root = render({ buttonType });
+    render({ buttonType: 'alert' });
 
-    expect(root.find(Button)).toHaveProp('buttonType', buttonType);
+    const button = screen.getByRole('button', {
+      name: defaultChildText,
+    });
+    expect(button).toHaveClass('Button--alert');
   });
 
   it('passes the children prop to the button', () => {
     const children = 'Do you really want to delete this?';
-    const root = render({ children });
+    render({ children });
 
-    expect(root.find(Button).children()).toHaveText(children);
+    expect(screen.getByRole('button', { name: children })).toBeInTheDocument();
   });
 
   it('shows ConfirmationDialog when button is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
+    render();
 
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    expect(root.find(ConfirmationDialog)).toHaveLength(0);
+    expect(
+      screen.queryByClassName('ConfirmationDialog'),
```

Comment:
We should avoid using `getByClassName` unless it's the only way we can target something important. In this case, the fact that the warning message is not shown (asserted on the next line) is enough to verify that the dialog is hidden, so this can simply be removed.